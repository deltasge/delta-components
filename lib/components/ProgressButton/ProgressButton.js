'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _withStyles = require('@material-ui/core/styles/withStyles');

var _withStyles2 = _interopRequireDefault(_withStyles);

var _Button = require('@material-ui/core/Button');

var _Button2 = _interopRequireDefault(_Button);

var _CircularProgress = require('@material-ui/core/CircularProgress');

var _CircularProgress2 = _interopRequireDefault(_CircularProgress);

var _styles = require('./styles');

var _styles2 = _interopRequireDefault(_styles);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ProgressButton = function ProgressButton(_ref) {
  var classes = _ref.classes,
      loading = _ref.loading,
      type = _ref.type,
      label = _ref.label,
      variant = _ref.variant,
      color = _ref.color,
      disabled = _ref.disabled,
      component = _ref.component,
      htmlFor = _ref.htmlFor,
      _ref$onClick = _ref.onClick,
      onClick = _ref$onClick === undefined ? null : _ref$onClick;
  return _react2.default.createElement(
    'div',
    { className: classes.rootButton },
    _react2.default.createElement(
      'div',
      { className: classes.wrapperButton },
      _react2.default.createElement(
        _Button2.default,
        { type: type, disabled: loading || disabled, variant: variant, color: color, className: classes.button, component: component, onClick: onClick, htmlFor: htmlFor },
        label
      ),
      loading && _react2.default.createElement(_CircularProgress2.default, { size: 24, className: classes.buttonProgress })
    )
  );
};

ProgressButton.propTypes = {
  /** Loading */
  loading: _propTypes2.default.bool,
  /** Desabilita o botão */
  disabled: _propTypes2.default.bool,
  /** Tipo do Botão */
  type: _propTypes2.default.string.isRequired,
  /** Label do Botão */
  label: _propTypes2.default.string.isRequired,
  /** Variação do Botão */
  variant: _propTypes2.default.oneOf(['text', 'flat', 'outlined', 'contained', 'raised', 'fab', 'extendedFab']),
  /** Cor do botão */
  color: _propTypes2.default.oneOf(['default', 'inherit', 'primary', 'secondary']),
  /** Componente */
  component: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.func, _propTypes2.default.object])
};

ProgressButton.defaultProps = {
  color: 'default',
  disabled: false,
  type: 'button',
  variant: 'text',
  component: 'button'
};

exports.default = (0, _withStyles2.default)(_styles2.default)(ProgressButton);