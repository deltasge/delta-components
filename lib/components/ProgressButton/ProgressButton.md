```
const { Button } = require('@material-ui/core');

initialState = {
  loading: false
};

reload = () => {
  const { loading } = state
  setState({ loading: !loading })
}

<div>
  <ProgressButton variant='contained' color='secondary' loading={state.loading} label='Label' type='button' />
  <Button variant='contained' onClick={reload}>Loading</Button>
</div>
```
