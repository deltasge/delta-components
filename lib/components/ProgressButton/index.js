'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _ProgressButton = require('./ProgressButton');

Object.defineProperty(exports, 'default', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_ProgressButton).default;
  }
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }