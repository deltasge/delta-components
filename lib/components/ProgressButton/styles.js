'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _blue = require('@material-ui/core/colors/blue');

var _blue2 = _interopRequireDefault(_blue);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function (theme) {
  return {
    rootButton: {
      display: 'flex',
      alignItems: 'center'
    },
    wrapperButton: {
      position: 'relative'
    },
    buttonProgress: {
      color: _blue2.default[500],
      position: 'absolute',
      top: '50%',
      left: '50%',
      marginTop: -12,
      marginLeft: -12
    },
    button: {
      float: 'right'
    }
  };
};