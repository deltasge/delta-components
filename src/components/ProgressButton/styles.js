import blue from '@material-ui/core/colors/blue'

export default (theme) => ({
  rootButton: {
    display: 'flex',
    alignItems: 'center'
  },
  wrapperButton: {
    position: 'relative'
  },
  buttonProgress: {
    color: blue[500],
    position: 'absolute',
    top: '50%',
    left: '50%',
    marginTop: -12,
    marginLeft: -12
  },
  button: {
    float: 'right'
  }
})
