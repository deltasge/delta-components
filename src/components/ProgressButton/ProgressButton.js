import React from 'react'
import PropTypes from 'prop-types'
import withStyles from '@material-ui/core/styles/withStyles'

import Button from '@material-ui/core/Button'
import CircularProgress from '@material-ui/core/CircularProgress'

import styles from './styles'

const ProgressButton = ({ classes, loading, type, label, variant, color, disabled, component, htmlFor, onClick = null }) => (
  <div className={classes.rootButton}>
    <div className={classes.wrapperButton}>
      <Button type={type} disabled={loading || disabled} variant={variant} color={color} className={classes.button} component={component} onClick={onClick} htmlFor={htmlFor}>{label}</Button>
      {loading && <CircularProgress size={24} className={classes.buttonProgress} />}
    </div>
  </div>
)

ProgressButton.propTypes = {
  /** Loading */
  loading: PropTypes.bool,
  /** Desabilita o botão */
  disabled: PropTypes.bool,
  /** Tipo do Botão */
  type: PropTypes.string.isRequired,
  /** Label do Botão */
  label: PropTypes.string.isRequired,
  /** Variação do Botão */
  variant: PropTypes.oneOf(['text', 'flat', 'outlined', 'contained', 'raised', 'fab', 'extendedFab']),
  /** Cor do botão */
  color: PropTypes.oneOf(['default', 'inherit', 'primary', 'secondary']),
  /** Componente */
  component: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object])
}

ProgressButton.defaultProps = {
  color: 'default',
  disabled: false,
  type: 'button',
  variant: 'text',
  component: 'button'
}

export default withStyles(styles)(ProgressButton)
