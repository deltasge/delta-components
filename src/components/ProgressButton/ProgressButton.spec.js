/* global describe it expect */

import React from 'react'
import ProgressButton from './index'
import { mount } from 'enzyme'

describe('<ProgressButton />', () => {
  it('should mount', () => {
    const wrapper = mount(<ProgressButton type='button' label='Label' />)
    expect(wrapper.containsMatchingElement(<span>Label</span>)).toEqual(true)
  })
  it('matches the snapshot', () => {
    const tree = mount(<ProgressButton type='button' label='Label' />)
    expect(tree).toMatchSnapshot()
  })
  it('should mount disabled button', () => {
    const wrapper = mount(<ProgressButton type='button' label='Label' disabled />)
    expect(wrapper.containsMatchingElement(<button type='button' disabled><span>Label</span></button>)).toEqual(true)
  })
  it('should mount loading button', () => {
    const wrapper = mount(<ProgressButton type='button' label='Label' loading />)
    expect(wrapper.find('circle').length).toEqual(1)
  })
})
