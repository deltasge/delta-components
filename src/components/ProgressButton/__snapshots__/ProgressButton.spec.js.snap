// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressButton /> matches the snapshot 1`] = `
<WithStyles(ProgressButton)
  label="Label"
  type="button"
>
  <ProgressButton
    classes={
      Object {
        "button": "ProgressButton-button-4",
        "buttonProgress": "ProgressButton-buttonProgress-3",
        "rootButton": "ProgressButton-rootButton-1",
        "wrapperButton": "ProgressButton-wrapperButton-2",
      }
    }
    color="default"
    component="button"
    disabled={false}
    label="Label"
    type="button"
    variant="text"
  >
    <div
      className="ProgressButton-rootButton-1"
    >
      <div
        className="ProgressButton-wrapperButton-2"
      >
        <WithStyles(Button)
          className="ProgressButton-button-4"
          color="default"
          component="button"
          disabled={false}
          onClick={null}
          type="button"
          variant="text"
        >
          <Button
            className="ProgressButton-button-4"
            classes={
              Object {
                "colorInherit": "MuiButton-colorInherit-26",
                "contained": "MuiButton-contained-16",
                "containedPrimary": "MuiButton-containedPrimary-17",
                "containedSecondary": "MuiButton-containedSecondary-18",
                "disabled": "MuiButton-disabled-25",
                "extendedFab": "MuiButton-extendedFab-23",
                "fab": "MuiButton-fab-22",
                "flat": "MuiButton-flat-10",
                "flatPrimary": "MuiButton-flatPrimary-11",
                "flatSecondary": "MuiButton-flatSecondary-12",
                "focusVisible": "MuiButton-focusVisible-24",
                "fullWidth": "MuiButton-fullWidth-30",
                "label": "MuiButton-label-6",
                "mini": "MuiButton-mini-27",
                "outlined": "MuiButton-outlined-13",
                "outlinedPrimary": "MuiButton-outlinedPrimary-14",
                "outlinedSecondary": "MuiButton-outlinedSecondary-15",
                "raised": "MuiButton-raised-19",
                "raisedPrimary": "MuiButton-raisedPrimary-20",
                "raisedSecondary": "MuiButton-raisedSecondary-21",
                "root": "MuiButton-root-5",
                "sizeLarge": "MuiButton-sizeLarge-29",
                "sizeSmall": "MuiButton-sizeSmall-28",
                "text": "MuiButton-text-7",
                "textPrimary": "MuiButton-textPrimary-8",
                "textSecondary": "MuiButton-textSecondary-9",
              }
            }
            color="default"
            component="button"
            disableFocusRipple={false}
            disabled={false}
            fullWidth={false}
            mini={false}
            onClick={null}
            size="medium"
            type="button"
            variant="text"
          >
            <WithStyles(ButtonBase)
              className="MuiButton-root-5 MuiButton-text-7 MuiButton-flat-10 ProgressButton-button-4"
              component="button"
              disabled={false}
              focusRipple={true}
              focusVisibleClassName="MuiButton-focusVisible-24"
              onClick={null}
              type="button"
            >
              <ButtonBase
                centerRipple={false}
                className="MuiButton-root-5 MuiButton-text-7 MuiButton-flat-10 ProgressButton-button-4"
                classes={
                  Object {
                    "disabled": "MuiButtonBase-disabled-32",
                    "focusVisible": "MuiButtonBase-focusVisible-33",
                    "root": "MuiButtonBase-root-31",
                  }
                }
                component="button"
                disableRipple={false}
                disableTouchRipple={false}
                disabled={false}
                focusRipple={true}
                focusVisibleClassName="MuiButton-focusVisible-24"
                onClick={null}
                tabIndex="0"
                type="button"
              >
                <button
                  className="MuiButtonBase-root-31 MuiButton-root-5 MuiButton-text-7 MuiButton-flat-10 ProgressButton-button-4"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={null}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiButton-label-6"
                  >
                    Label
                  </span>
                  <NoSsr
                    defer={false}
                    fallback={null}
                  >
                    <WithStyles(TouchRipple)
                      center={false}
                      innerRef={[Function]}
                    >
                      <TouchRipple
                        center={false}
                        classes={
                          Object {
                            "child": "MuiTouchRipple-child-38",
                            "childLeaving": "MuiTouchRipple-childLeaving-39",
                            "childPulsate": "MuiTouchRipple-childPulsate-40",
                            "ripple": "MuiTouchRipple-ripple-35",
                            "ripplePulsate": "MuiTouchRipple-ripplePulsate-37",
                            "rippleVisible": "MuiTouchRipple-rippleVisible-36",
                            "root": "MuiTouchRipple-root-34",
                          }
                        }
                      >
                        <TransitionGroup
                          childFactory={[Function]}
                          className="MuiTouchRipple-root-34"
                          component="span"
                          enter={true}
                          exit={true}
                        >
                          <span
                            className="MuiTouchRipple-root-34"
                          />
                        </TransitionGroup>
                      </TouchRipple>
                    </WithStyles(TouchRipple)>
                  </NoSsr>
                </button>
              </ButtonBase>
            </WithStyles(ButtonBase)>
          </Button>
        </WithStyles(Button)>
      </div>
    </div>
  </ProgressButton>
</WithStyles(ProgressButton)>
`;
